packages/components/src/data-model/word-query/word-query.js

```diff
@@ -0,0 +1,176 @@
+import { Logger } from 'alpheios-data-models'
+import TuftsMorphologyData from '@comp/data-model/word-query/lexical-data/data-objects/tufts-morphology-data.js'
+import TreebankData from '@comp/data-model/word-query/lexical-data/data-objects/treebank-data.js'
+import WordAsLexemeData from '@comp/data-model/word-query/lexical-data/data-objects/word-as-lexeme-data.js'
+import DisambiguatedData from '@comp/data-model/word-query/lexical-data/data-objects/disambiguated-data.js'
+import ShortDefsData from '@comp/data-model/word-query/lexical-data/data-objects/short-defs-data.js'
+
+/**
+ * Contains all the business logic that is necessary to resolve the GraphQL word query.
+ */
+export default class WordQuery {
+  constructor ({
+    word,
+    language,
+    clientId,
+    getLexemes,
+    useMorphService,
+    useTreebankData,
+    useWordAsLexeme,
+    getShortDefs,
+    treebankProvider,
+    treebankSentenceId,
+    treebankWordIds,
+    lexicons,
+    shortLexicons,
+    wordQueryResponse
+  }) {
```

Comment:
Don't like the long list of parameters but has no idea how to avoid using it.

Reply:
perhaps we should subdivide them into logical groups?

useMorphService, useTreebankData useWordAsLexeme are all only relevant for getLexemes so they could be properties on the getLexemes option

treebankProvider, treebankSentenceId, treebankWord ids could be subproperties on useTreebankData

lexicons, shortLexicons could be subproperties on getShortDefs

Reply:
Seen your response later on the page but not in this place. The second one was without an ability to reply so I cited it in a separate message at the very bottom.

I think it's a great idea and I will implement the suggested change.